docs/ref/models/expressions.txt

```diff
@@ -588,22 +588,38 @@ expressions::
 
 Next, we write the method responsible for generating the SQL::
 
-    def as_sql(self, compiler, connection):
+    def as_sql(self, compiler, connection, **kwargs):
         sql_expressions, sql_params = [], []
         for expression in self.expressions:
             sql, params = compiler.compile(expression)
             sql_expressions.append(sql)
             sql_params.extend(params)
         self.extra['expressions'] = ','.join(sql_expressions)
-        return self.template % self.extra, sql_params
+        template = kwargs.get('template', self.template)
+        return template % self.extra, sql_params
 
-    def as_oracle(self, compiler, connection):
+    def as_oracle(self, compiler, connection, **kwargs):
         """
         Example of vendor specific handling (Oracle in this case).
         Let's make the function name lowercase.
         """
-        self.template = 'coalesce( %(expressions)s )'
-        return self.as_sql(compiler, connection)
+        return self.as_sql(compiler, connection,
+            template='coalesce( %(expressions)s )'
+        )
+
+.. versionchanged:: 1.10
+
+    ``as_sql``-methods now support ``kwargs`` to override the default
+    ``template``- or ``function``-variables of the ``Expression``-class. This
+    is useful for 3rd-party database-backends that may have different
+    SQL-syntax. Simply override ``as_sql`` with your backend-specific
+    ``as_vendorname``-method and call the existing ``as_sql``-method with the
+    modified ``template``- or ``function``-parameter.
```

Comment:
As above, rewrite as:

... "with the modified ``template`` or context parameters."